### PR TITLE
🏃 Add verification tools, linter and modules

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

This PR cleans up the use of linters, uses Go Modules to build specific version of external tools (golangci-lint), and provides a way to check that go modules are always up to date.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-#  Copyright 2019 The Kubernetes Authors.
+#  Copyright 2020 The Kubernetes Authors.
```
Comment (shawn-hurley) [inline Makefile]: more for my own knowledge, do we usually update copyright year in kubernetes when a file is updated?

**Comment by hypnoglow** [comment]
It seems `hack/tools/hack/tools/bin/golangci-lint` binary is committed unintentionally.

The rest LGTM

**Comment by vincepri** [comment]
@hypnoglow I force pushed and removed it, I also don't see it in the file list, am I missing something?

File: Makefile (hunk 1 above)
Comment (vincepri) [reply Makefile]: I don't have a good answer, I changed it because it was mostly rewritten

**Comment by vincepri** [comment]
/assign @alenkacz @gerred 

would love to get more 👀 and feedback on this one

**Comment by gerred** [comment]
:+1: I'll take a look - was out sick for the past few days

**Comment by alenkacz** [comment]
/lgtm

this actually looks great, love the cleaned up makefile. Should we update some documentation or use the new targets somewhere? 🤔

**Pull request merged**